image/config.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package image
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/opencontainers/image-spec/schema"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+)
+
+type cfg struct {
+	User         string
+	Memory       int64
+	MemorySwap   int64
+	CPUShares    int64 `json:"CpuShares"`
+	ExposedPorts map[string]struct{}
```

Comment:
On Thu, Jun 02, 2016 at 11:49:50AM -0700, Vincent Batts wrote:
> > +	ExposedPorts map[string]struct{}
> 
> But i'm sure that other extentsions will come along and use such.

How do they find it?  If this command unpacks $IMAGE into $BUNDLE, but
does not leave a pointer in $BUNDLE to say “this was unpacked from
$IMAGE”, I don't see how tooling invoked on the $BUNDLE would be able
to find this image-specific data.  So to support that additional
tooling, you either need:

a. A generic “this was unpacked from $IMAGE” reference in the bundle
   directory somewhere, or
b. To copy all of the “unused” information like ExposedPorts over to
   some well-known location inside the bundle.